src/pages/SupplyRequisitionPage.js

```diff
@@ -0,0 +1,362 @@
+            options={this.props.database.objects('Item')}
+            queryString={'name BEGINSWITH[c] $0 OR code BEGINSWITH[c] $0'}
+            queryStringSecondary={'name CONTAINS[c] $0'}
+            sortByString={'name'}
+            onSelect={(item) => {
+              const { database, requisition } = this.props;
+              database.write(() => {
+                if (!requisition.hasItemWithId(item.id)) {
+                  createRecord(database, 'RequisitionItem', requisition, item);
+                  database.save('Requisition', requisition);
+                }
+              });
+              this.refreshData();
+              this.closeModal();
+            }}
+            renderLeftText={(item) => `${item.name}`}
+            renderRightText={(item) => `${item.totalQuantity}`}
+          />
+        );
+      case COMMENT_EDIT:
+        return (
+          <TextEditor
+            text={this.props.requisition.comment}
+            onEndEditing={(newComment) => {
+              if (newComment !== this.props.requisition.comment) {
+                this.props.database.write(() => {
+                  this.props.requisition.comment = newComment;
+                  this.props.database.save('Requisition', this.props.requisition);
+                });
+              }
+              this.closeModal();
+            }}
+          />
+          );
+    }
+  }
+
+  renderButtons() {
+    return (
+      <View style={globalStyles.pageTopRightSectionContainer}>
+        <View style={globalStyles.verticalContainer}>
+          <PageButton
+            style={globalStyles.topButton}
+            text={buttonStrings.use_requested_quantity}
+            onPress={this.onUseRequestedQuantity}
+            isDisabled={this.props.requisition.isFinalised}
+          />
+        </View>
+      </View>
+    );
+  }
+
+  render() {
+    return (
+      <GenericPage
+        data={this.state.data}
+        refreshData={this.refreshData}
+        renderCell={this.renderCell}
+        renderTopLeftComponent={this.renderPageInfo}
+        renderTopRightComponent={this.renderButtons}
+        onEndEditing={this.onEndEditing}
+        onSelectionChange={this.onSelectionChange}
+        defaultSortKey={this.dataFilters.sortBy}
+        defaultSortDirection={this.dataFilters.isAscending ? 'ascending' : 'descending'}
+        columns={[
+          {
+            key: 'itemCode',
+            width: 1.5,
+            title: tableStrings.code,
+            sortable: true,
+          },
+          {
+            key: 'itemName',
+            width: 4,
+            title: tableStrings.item_name,
+            sortable: true,
+          },
+          {
+            key: 'ourStockOnHand', // TODO: should this column name be consistent with other pages
+            width: 1.5,
+            title: tableStrings.our_stock,
+            sortable: true,
+            alignText: 'right',
+          },
+          {
+            key: 'stockOnHand',
```

Comment:
I am glad you mentioned this actually. Request requisition.stockOnHand should be 'dynamic' until finalised and update (alongside suggestedQuantity), not set in createRecord. I have other thought on this subject, will try to update by mid Tomorrow otherwise will make an issue.